### PR TITLE
PWX-30419: Use reconcile loop to check pre-flight status. Timeout after 15m.

### PR DESCRIPTION
**Pull request by jrivera-px**
**What this PR does / why we need it**:
When executing pre-flight check we would wait for the checks to complete blocking the reconcile loop. This prevented any other state change from being processed. So this change corrected that. Use cluster conditions to determine if we are still executing a pre-flight. If so check the status each time though the loop. Add a 15 min timeout based on the time the DS has been running.
**Which issue(s) this PR fixes** (optional)
Closes #
PWX-30419
**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/controller/storagecluster/storagecluster.go [hunk 1]
```diff
@@ -412,55 +412,59 @@ func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster
 	return nil
 }
 
-func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
+func bringUpStorageCluster(cluster *corev1.StorageCluster) (bool, error) {
```
Comment (piyush-nimbalkar) [inline pkg/controller/storagecluster/storagecluster.go:415]: We don't need to return an error here.

File: pkg/controller/storagecluster/storagecluster.go [hunk 2]
```diff
@@ -657,6 +714,15 @@ func (c *Controller) syncStorageCluster(
 		}
 	}
 
+	if startOk, err := bringUpStorageCluster(cluster); err == nil {
```
Comment (piyush-nimbalkar) [inline pkg/controller/storagecluster/storagecluster.go:717]: nit: isPreflightComplete?

File: drivers/storage/portworx/portworx.go [hunk 3]
```diff
@@ -112,50 +130,61 @@ func (p *portworx) Validate(cluster *corev1.StorageCluster) error {
...
+	}
 
-		if total != 0 && completed == total {
-			logrus.Infof("pre-flight: completed...")
-			break
+	completed, inProgress, total, age, err := preFlighter.GetPreFlightStatus()
+	if err != nil {
+		if errors.IsNotFound(err) {
+			condition.Status = corev1.ClusterConditionStatusInProgress
+			condition.Message = "pre-flight: in progress..."
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:142]: *Preflight pods haven't started yet.

**Review by piyush-nimbalkar — APPROVED** [review]
@jrivera-px can we UTs for these changes either in this PR or a different one?

File: pkg/controller/storagecluster/storagecluster.go [hunk 4]
```diff
@@ -842,7 +920,7 @@ func (c *Controller) gcNeeded(obj client.Object) bool {
 
 	if obj.GetObjectKind().GroupVersionKind().Kind == "ConfigMap" {
 		if obj.GetName() == "px-attach-driveset-lock" ||
-			strings.HasPrefix(obj.GetName(), "px-bringup-queue-lock") {
+			strings.HasPrefix(obj.GetName(), "pxb-ringup-queue-lock") {
```
[the remark this answers is not in the document]
Comment (jrivera-px) [reply pkg/controller/storagecluster/storagecluster.go:923]: yeah transpose the chars on accident..

**Pull request merged**